jmeter-java-dsl/src/test/java/us/abstracta/jmeter/javadsl/core/BuildTreeContextTest.java

```diff
@@ -0,0 +1,55 @@
+package us.abstracta.jmeter.javadsl.core;
+
+import org.junit.jupiter.api.Test;
+import static org.junit.jupiter.api.Assertions.*;
+
+import java.io.File;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.LinkedHashMap;
+
+public class BuildTreeContextTest {
+
+    @Test
+    public void setEntryAddsKeyValuePair() {
+        BuildTreeContext context = new BuildTreeContext();
+        String key = "key";
+        String value = "value";
+
+        context.setEntry(key, value);
+
+        assertEquals(value, context.getEntry(key));
+    }
+
+    @Test
+    public void buildRemoteExecutionContextInitializesMaps() {
+        BuildTreeContext context = BuildTreeContext.buildRemoteExecutionContext();
+
+        assertNotNull(context.getVisualizers());
+        assertTrue(context.getVisualizers().isEmpty());
+        assertNotNull(context.getAssetFiles());
+        assertTrue(context.getAssetFiles().isEmpty());
+    }
+
+    @Test
+    public void processAssetFileHandlesDuplicatesByPrependingIndex() throws Exception {
+        BuildTreeContext context = BuildTreeContext.buildRemoteExecutionContext();
+        setAssetFilesField(context, new LinkedHashMap<>());
+
+        Path tempFile1 = Files.createTempFile("sample", ".txt");
+        String processedPath1 = context.processAssetFile(tempFile1.toString());
+        assertEquals(tempFile1.getFileName().toString(), new File(processedPath1).getName());
+
+        String processedPath2 = context.processAssetFile(tempFile1.toString());
+
+        assertEquals(tempFile1.getFileName().toString(), processedPath2, "Processed file name should have a index for duplicates.");
```

Comment:
The test doesn't seem to test what the test name suggests, does it? Currently the test seems to test that if you process the same file twice, you get the same path. But the index is only prepended when files have same name, but are not the same file (filepath differs but filenames are equal).